docs/DeveloperGuide.md

```diff
@@ -6,7 +6,152 @@
+
+Here is the typical design of a manager:
+
+![](diagrams/class/manager.png)
+
+
+### Menu (dishes) Management
+
+The management of menu is handled by `DishManager`, which implements following methods:
+
+- `void printDishes()` -- print all dishes and their menu index in current menu.
+- `void addDish(Dish dish)` -- add a new dish.
+- `void deleteDish(int menuIndex)` -- delete a dish using its menu index.
+- `void setPrice(int menuIndex, double newPrice)` -- change the price of a dish using its menu index.
+- `void setName(int menuIndex, String newName)` -- change the name of a dish using its menu index.
+
+These functions are "delegated" to `ArrayList<Dish>`, which is associated with `DishManager`.
+
+For example, `printDishes()`, iterates through `dishes` and calls `get(int)` method (`setPrice(int, double)`, `setName(int, String)` also call `get(int)`):
+
+![](diagrams/sqeuence/DishManager_printDishes.png)
+
+`deleteDish(int)` calls `remove(int)`:
+
+![](diagrams/sqeuence/DishManager_deleteDish.png)
+
+`addDish(Dish)` calls `add(Dish)`:
+
+![](diagrams/sqeuence/DishManager_addDish.png)
+
+
+
+### Order Management
+
+The management of menu is functioned by `OrderManager`, which implements following methods:
+
+- `void addOrder(Order order)` -- add a new order.
+- `int addDishToOrder(Object inputObj)` -- add dishes in to an order.
+- `void deleteOrder(int orderIndex)` -- delete an order using its order index.
+- `Order getOrder(int orderIndex)` -- get an order using its order index.
+- `double getOrderPrice(int orderIndex)` -- get the total price of an order using its order index.
+- `double getAllOrderValue()` -- get the total price of all orders.
+
+These functions are "delegated" to `ArrayLise<Order>`, which is associated with `OrderManager`.
+
+For example, `getOrder(int)`, iterates through `orders` and calls `get(int)` method (`getOrderPrice(int)`, `getAllOrderValue()` also call `get(int)`):
+
+
+`deleteOrder(int)` calls `remove(int)`:
+
+
+`getOrderPrice(int)` calls `getTotalPrice()` and method `getAllOrderValue()` also calls `getTotalPrice()`:
+
+### Staff Management
+
+The management of staff is handled by `StaffManager`, which implements following methods:
+
+- `void printStaff()` -- print all staffs.
+- `void addStaff(int staffId, String staffName, String position, double salary)` -- add a new staff with the arguments as its attributes.
+- `Staff findByStaffId(int staffId, boolean printMsg)` -- find a staff using staff id, and print message if boolean variable printMsg is true.
+- `void deleteByStaffId(int staffId)` -- delete a staff using staff id.
+- `int getNumOfStaffs(` -- change the name of a dish using its menu index.
+
+`StaffManager` manages `ArrayList<Staff>` and the above functions help to update the collection of `Staff`.
+
+For example `addStaff(int staffId, String staffName, String position, double salary)` calls `add(Staff)` to add a `Staff` from `ArrayList<Staff>`:
+
+![](diagrams/sqeuence/StaffManager_addStaff.png)
```

Comment:
The font sizes are too small to read at normal zoom size, might want to increase the font size or increase diagram size.
![image](https://user-images.githubusercontent.com/65325165/160958080-e906304a-0eb3-4101-92ff-1d1d9826e792.png)